app/src/pages/CardSubmission/SubmissionForm.tsx

```diff
@@ -1,33 +1,54 @@
 import { Button, Grid } from "@material-ui/core"
 import { CurrentGameContext } from "contexts/CurrentGame"
-import { CurrentPlayerContext } from "contexts/CurrentPlayer"
+import { CurrentPlayerContext, PlayerRole } from "contexts/CurrentPlayer"
 import { useSubmitCardsMutation } from "generated/graphql"
-import { cloneDeep } from "lodash"
+import { cloneDeep, filter } from "lodash"
 import { Title } from "pages/CardSubmission"
 import SubmissionCard from "pages/CardSubmission/SubmissionCard"
 import * as React from "react"
 
 function SubmissionForm(props: { onSubmit: () => void }) {
-  const DEFAULT_NUM_ENTRIES = 3
   const currentPlayer = React.useContext(CurrentPlayerContext)
   const currentGame = React.useContext(CurrentGameContext)
   const [submitCards, { called }] = useSubmitCardsMutation()
+
+  const numSubmitted = filter(
+    currentGame.cards,
+    (card) => card.player_id === currentPlayer.id
+  ).length
+
+  const numToSubmit =
+    (currentGame.num_entries_per_player &&
+      currentGame.num_entries_per_player - numSubmitted) ||
+    0
+
   const [words, setWords] = React.useState<Array<string>>(
     Array.from(
-      { length: currentGame.num_entries_per_player || DEFAULT_NUM_ENTRIES },
+      {
+        length: numToSubmit,
+      },
       () => ""
     )
   )
 
+  React.useEffect(() => {
+    setWords(
+      Array.from(
+        {
+          length: numToSubmit,
+        },
+        () => ""
+      )
+    )
+  }, [numToSubmit])
```

Comment:
similar to above - make this form reactive to host accepting/rejecting.

edge cases: 1 of your cards gets rejected. then as you are filling out a new one, another one gets rejected. do we keep your other word and add on a new card? or just wipe it all and give you two fresh cards? etc.

Reply:
Resolved in https://github.com/avimoondra/fishbowl/pull/101